### PR TITLE
Add Kimi-k2-thinking and k2.5 and k2.6 checkpoint conversion support.

### PR DESCRIPTION
**Pull request by gagika**
# Description

  - Extends convert_deepseek_family_ckpt.py to convert HuggingFace Kimi checkpoints (kimi-k2-thinking, kimi-k2.5-text,      
  kimi-k2.6-text) to the MaxText DeepSeek family layout.
  - Adds an int4 dequantization helper (dequantize_pack_quantized_int4) for the routed-expert projections, which ship as    
  compressed-tensors pack-quantized weights with per-group symmetric scales (group_size=32).                                
  - K2.5 and K2.6 are extracted from their multimodal KimiK25ForConditionalGeneration wrapper — the text branch shape and quantization layout are identical, so K2.6 reuses the K2.5 key map.                                                       
                                                                                                                            
# Tests   
                                                                                                                            
- pytest tests/unit/dequantize_pack_quantized_int4_test.py
- Convert each variant end-to-end and verify forward-pass logits match HF on a single prompt
- Kimi k2.5 logit matching `KL divergence = [0.00043962 0.00289483]`: https://paste.googleplex.com/6351128208998400

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py [hunk 1]
```diff
@@ -232,10 +315,48 @@ def _convert_huggingface_to_jax_weights(base_model_path, model_params, mem_info,
...
+      return raw_key
+    if raw_key.startswith(hf_key_prefix):
+      return raw_key[len(hf_key_prefix) :]
+    return None
+
   for i, ckpt_path in enumerate(ckpt_paths):
     max_logging.log(f"Loading checkpoint {i+1} of {len(ckpt_paths)} ...")
     with safe_open(ckpt_path, framework="pt", device="cpu") as f:
-      for key in f.keys():
+      for raw_key in f.keys():
```
Comment (richjames0) [inline src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py:332]: This could be a bias because I hate continue statements but I reckon if you ran the cognitive complexity numbers of this it would come out high and it might be worth asking a bot if it can simplify

**Review by richjames0 — APPROVED** [review]
Great testing Gagik. One nth comment

File: src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py (hunk 1 above)
Comment (gagika) [reply src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py:332]: little bit simplified with an agent

File: src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py [hunk 2]
```diff
@@ -48,6 +48,39 @@
 absl.logging.set_verbosity(absl.logging.INFO)  # for max_logging.log
 
 
+# Group size used by Kimi-K2 quantized variants for per-group symmetric int4 scales.
+# Matches `quantization_config.config_groups.group_0.weights.group_size` in the HF config.
+INT4_GROUP_SIZE = 32
+
+
+def dequantize_pack_quantized_int4(
```
Comment (shuningjin) [inline src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py:56]: I guess this can run on CPU? Maybe could add a comment to clarify, as many other de-quantization needs GPU

**Review by shuningjin — APPROVED** [review]
Thank you for expanding support to multiple variants (K2-Thinking, K2.5, K2.6) on top of Kimi-K2! The int4 dequantization with unit tests, checkpoint conversion, and user guide updates all look great.

File: src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py (hunk 2 above)
Comment (gagika) [reply src/maxtext/checkpoint_conversion/standalone_scripts/convert_deepseek_family_ckpt.py:56]: Thanks, good point, I updated the comment.
I have excluded this test from CI / CD, as it uses PyTorch, but updated it to have CPU only tag.
